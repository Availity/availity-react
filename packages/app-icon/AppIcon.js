import React from 'react';
import PropTypes from 'prop-types';

function AppIcon({
  tag: Tag,
  color,
  size,
  src: image,
  alt,
  branded,
  className,
  children,
  ...props
}) {
  const classname = [
    className,
    'app-icon',
    branded ? `app-icon-branded-${color}` : `app-icon-${color}`,
    size && `app-icon-${size}`,
    image && 'border-0',
  ]
    .filter(a => a)
    .join(' ');

  return (
    <Tag {...props} className={classname}>
      {image ? (
        <img className="w-100 h-100 align-baseline" src={image} alt={alt} />
      ) : (
        children
      )}
      {branded && <span className="caret" />}
    </Tag>
  );
}

AppIcon.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  branded: PropTypes.bool,
  className: PropTypes.string,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  children: PropTypes.node,
  src: PropTypes.string,
  alt: PropTypes.string,
};

AppIcon.defaultProps = {
  tag: 'span',
  color: 'black',
};

export default AppIcon;
