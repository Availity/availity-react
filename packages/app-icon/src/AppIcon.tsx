import React from 'react';
import classNames from 'classnames';

type Props = {
  alt?: string;
  branded?: boolean;
  children?: React.ReactNode;
  className?: string;
  color?: string;
  size?: string;
  src?: string;
  tag?: React.ElementType;
};

const AppIcon = ({
  alt,
  branded,
  children,
  className,
  color = 'black',
  size,
  src: image,
  tag: Tag = 'span',
  // TODO: look into removing this and officially accepting the props
  ...props
}: Props): JSX.Element => {
  const classes = classNames(className, 'app-icon', {
    [`app-icon-${color}`]: color && !branded,
    [`app-icon-branded-${color}`]: color && branded,
    [`app-icon-${size}`]: size,
    'border-0': image,
  });

  return (
    <Tag {...props} className={classes}>
      {image ? <img className="w-100 h-100 align-baseline" src={image} alt={alt} /> : children}
      {branded ? <span className="caret" /> : null}
    </Tag>
  );
};

export default AppIcon;
