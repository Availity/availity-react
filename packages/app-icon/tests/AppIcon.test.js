import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AppIcon from '..';

afterEach(cleanup);

describe('AppIcon', () => {
  test('should render', () => {
    const { container } = render(<AppIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render color', () => {
    const { container } = render(<AppIcon color="green" />);

    expect(container.firstChild.className).toContain('app-icon-green');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render size', () => {
    const { container } = render(<AppIcon size="lg" />);

    expect(container.firstChild.className).toContain('app-icon-lg');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render children', () => {
    const { container } = render(<AppIcon>AI</AppIcon>);

    expect(container.firstChild.textContent).toEqual('AI');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render branded', () => {
    const { container } = render(<AppIcon branded />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render additional classNames', () => {
    const { container } = render(<AppIcon className="and-more" />);

    expect(container.firstChild.className).toContain('and-more');
    expect(container).toMatchSnapshot();
  });

  test('should render additional attributes', () => {
    const { container, getAllByTitle } = render(
      <AppIcon title="Availity Icon" />
    );

    expect(getAllByTitle('Availity Icon')).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  test('should render custom tags', () => {
    const { container } = render(<AppIcon tag="i" />);

    expect(container.firstChild.tagName.toLowerCase()).toEqual('i');
    expect(container).toMatchSnapshot();
  });

  test('should render with image', () => {
    const { container } = render(
      <AppIcon src="/path-to-some-image" alt="icon" />
    );
    const image = container.querySelectorAll('img')[0];
    expect(image.src).toEqual('http://localhost/path-to-some-image');
  });
});
