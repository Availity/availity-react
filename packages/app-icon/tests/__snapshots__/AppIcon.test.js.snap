// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon should render 1`] = `
<span
  class="app-icon app-icon-black"
/>
`;

exports[`AppIcon should render additional attributes 1`] = `
<div>
  <span
    class="app-icon app-icon-black"
    title="Availity Icon"
  />
</div>
`;

exports[`AppIcon should render additional classNames 1`] = `
<div>
  <span
    class="and-more app-icon app-icon-black"
  />
</div>
`;

exports[`AppIcon should render branded 1`] = `
<span
  class="app-icon app-icon-branded-black"
>
  <span
    class="caret"
  />
</span>
`;

exports[`AppIcon should render children 1`] = `
<span
  class="app-icon app-icon-black"
>
  AI
</span>
`;

exports[`AppIcon should render color 1`] = `
<span
  class="app-icon app-icon-green"
/>
`;

exports[`AppIcon should render custom tags 1`] = `
<div>
  <i
    class="app-icon app-icon-black"
  />
</div>
`;

exports[`AppIcon should render size 1`] = `
<span
  class="app-icon app-icon-black app-icon-lg"
/>
`;
